Enterprise-CMCS/cmcs-eregulations
# EREGCSC-1957 -- Statute switcher list [CLEAN]

Resolves [EREGCSC-1957](https://jiraent.cms.gov/browse/EREGCSC-1957) and [EREGCSC-1923](https://jiraent.cms.gov/browse/EREGCSC-1923)

**Description**

This pull request includes all work done in the `1923-statute-tune-up` branch.  The [pull request for that story](1923-statute-tune-up) was approved, but closed without merging to `main` because additional stories related to the statute table were not yet ready. 

This pull request that you are reading right now will contain all work from 1923 as well as new work for 1957.

New features and functionality specific to this pull request:

- Adds list of SSA Titles that, when clicked, will refine the table to only display rows from that title.  The default title selection is SSA Title XIX / 19.

**This pull request changes:**

- Adds a `StatuteSelector` component to control which act and title is displayed on the Statutes page
- Styles `StatuteSelector` to match [Figma](https://www.figma.com/proto/mXCFChrIR50gbFAvXV0kgB/statute-to-USC-conversion?type=design&node-id=60-618&scaling=min-zoom&page-id=60%3A617)
- Adds Vitest tests for `StatuteSelector` component and extends existing Cypress end to end tests for the Statutes page
- Updates the table caption based on Slack conversation with Britta

**Steps to manually verify this change:**

1. Visit [experimental deployment](https://lyehc1jclf.execute-api.us-east-1.amazonaws.com/dev901) and make sure it looks like the Figma doc and matches the behavior notes (except the horizontal scrolling behavior, which is not implemented)
2. Make sure all Vitest and Cypress tests pass
